ad7768-1fmcz/ad77681.c

```diff
@@ -446,23 +1109,761 @@ int32_t ad77681_soft_reset(struct ad77681_dev *dev)
 }
 
 /**
- * Initialize the device.
+ * Device reset over SPI.
+ * @param dev 			The device structure.
+ * @param rst_option	Choose between soft and hard reset
+ *						Accepted values: AD77681_SOFT_RESET
+ *										 AD77681_HARD_RESET
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad77681_reset(struct ad77681_dev *dev,
+	enum ad7761_reset_option rst_option)
+{
+	int32_t ret = 0;
+
+	if (rst_option == AD77681_SOFT_RESET)							// Soft reset over SPI
+	{		
+		ret |= 	ad77681_spi_write_mask(dev,							// Two writes are required to initialize the reset
```

Comment:
Here you should initialize directly the ret variable, without performing the `or` operation.